Implement InfoButton for ORKTextChoice

eschramm opened the pull request:
Closes https://github.com/CareEvolution/RKStudio-Participant/issues/668. Implemented for both ORK1Kit and ResearchKit (RK2).

Note that there already is a `detailText` available on `ORKTextChoice`. Using the three options for `ORKChoiceDescriptionStyle`, the following behavior is observed.
- ORKChoiceDescriptionStyleNone - no `detailText` will show
- ORKChoiceDescriptionStyleDisplayAlways - if `detailText` exists, it will show
- ORKChoiceDescriptionStyleDisplayWhenExpanded - if 'detailText` exists, initially an info button will appear on the trailing side of the text with no `detailText` showing. When tapped, it will animate the expansion of the `detailText` into the cell.

In RK1, the Done/Next button is manually drawn in the `ORK1TableContainerView` and when the cell grows to show the `detailText`, the Done/Next button can inconveniently be pushed below the visible content in the scrollView. This also happens today with the hidePredicate in `ORKFormItemViewController. A solution is implemented here which sends the expected change in size to the TableContainerView which can adjust the constraint anchoring the button. Because this animation is independent of the TableView animation, it initially created a jumping with the tableView animation and then sliding back to expected position. Here we hide it at resize and then animate into view after the TableView resize to minimize the jumping.

Behavior was implemented for both the `ORK(1)QuestionViewController` and the `ORK(1)FormViewController` and tested with hidePredicate interaction as well. Tests added for basic behavior in CEVResearchKit.

![standardRK1](https://user-images.githubusercontent.com/1008462/77268501-291d1780-6c74-11ea-9197-1282525d90a7.gif)

In RK2, the animation of the TableView looks particularly bad if collapsing a large amount of detailText. I started to attempt to convert to auto-layout, thinking it might help and not only was it rather difficult to duplicate the style, it would be even more difficult to animate more cleanly due to being centered vertically when `detailText` is hidden, but top-justified when showing. I gave up for now and if we would rather have in NOT animate, we can simply reload the tableViewCell instead of adjusting it.

![rk2](https://user-images.githubusercontent.com/1008462/77268376-ac8a3900-6c73-11ea-89de-b92482b743d0.gif)

Also, in reviewing warnings, I noticed the recent [Stroop update](#55) did not completely implement NSCopying, omitting `copyWithZone:` so I added that here.

eschramm committed:
implement infoButton for detailText on ORKTextChoice

eschramm committed:
fix warning - NSCopying copyWithZone missing for Stroop classes

eschramm committed:
fix default state to ensure displayWhenExpanded defaults closed

mmertsock commented on ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.m:
```diff
@@ -1039,7 +1041,8 @@ - (BOOL)isEqual:(id)object {
             && ORK1EqualObjects(self.value, castObject.value)
-            && self.exclusive == castObject.exclusive);
+            && self.exclusive == castObject.exclusive
+            && self.detailTextShouldDisplay == castObject.detailTextShouldDisplay);
```
I'm not sure what the standard behavior is in ResearchKit: since `detailTextShouldDisplay` can change at runtime based on UI interactions, should it be included in isEqual and encode/decode operations?

eschramm replied:
Good point - adjusted

mmertsock commented on ORK1Kit/ORK1Kit/Common/ORK1Types.h:
```diff
@@ -140,6 +140,25 @@ typedef NS_ENUM(NSInteger, ORK1ChoiceAnswerStyle) {
+    /**
+     The detailText/description always appears under the answer choice
+     */
+    ORK1ChoiceDescriptionStyleDisplayAlways,
```
Would it make more sense for Always, instead of None, to be the default (zero) value?

eschramm replied:
I agree - good catch

mmertsock reviewed (commented):
I think your approach for addressing layout changes is fine. I haven't tested it on a device yet — do you know if that 15pt wide button is a large enough hit target for a touchscreen? Could make it a larger target by increasing the button's frame size, decrease the padding around it, and add some insets to the button to keep the `(i)` centered

eschramm committed:
PR improvements

- add padding around image for detailTextIndicator button so touch target is larger
- adjusted NSCopying for ORKTextChoice based on dynamic nature of detailTextShouldDisplay
- rearrange Enum so default is ORKChoiceDescriptionStyleDisplayAlways

eschramm commented:
Thanks for the review. That makes sense - I adjusted the touch target to be 30 x 30 points using inset to keep the image at 15 x 15.

mmertsock reviewed (approved):
LGTM 🚢
